srtcore/api.cpp

```diff
@@ -1520,11 +1520,20 @@ void CUDTUnited::removeSocket(const SRTSOCKET u)
       for (set<SRTSOCKET>::iterator q = i->second->m_pQueuedSockets->begin();
          q != i->second->m_pQueuedSockets->end(); ++ q)
       {
-         m_Sockets[*q]->m_pUDT->m_bBroken = true;
-         m_Sockets[*q]->m_pUDT->close();
-         m_Sockets[*q]->m_tsClosureTimeStamp = srt::sync::steady_clock::now();
-         m_Sockets[*q]->m_Status = SRTS_CLOSED;
-         m_ClosedSockets[*q] = m_Sockets[*q];
+         sockets_t::iterator si = m_Sockets.find(*q);
+         if (si == m_Sockets.end())
```

Comment:
There's a small bugfix here. Previous version was blindly addressing `m_Sockets[*q]` without checking for existence (so, potentially adding an element). The changed version simply does nothing when the element is not there and reports an IPE. This should rather never happen, but it covers a potential execution path that may lead to a crash.